### PR TITLE
Feature/dss10 api evaluate recipe creation

### PR DESCRIPTION
**Pull request by vthorey**
[[ch58502]](https://app.clubhouse.io/dataiku/story/58502/python-api-client-for-evaluate-recipe-creation)

- Add Evaluation recipe Creator

Companion backend PR: https://github.com/dataiku/dip/pull/13216

### CODE REVIEW

File: dataikuapi/dss/recipe.py [hunk 1]
```diff
@@ -1313,6 +1314,60 @@ def with_input_model(self, model_id):
...
+        builder.with_input_model(saved_model_id)
+        builder.with_input("dataset_to_evaluate")
+
+        builder.with_output("output_scored")
+        builder.with_output_metrics("output_metrics")
+        builder.with_output_evaluation_store(evaluation_store_id)
+
+        new_recipe = builder.build()
+    
+    Outputs can be created with
```
Comment (lpenet) [inline dataikuapi/dss/recipe.py:1337]: I would be more assertive. Something like:
"Outputs must exists. You can create them with"

File: dataikuapi/dss/recipe.py [hunk 2]
```diff
@@ -1313,6 +1314,60 @@ def with_input_model(self, model_id):
...
+
+        builder = project.new_managed_dataset("output_scored")
+        builder.with_store_into(connection)
+        dataset = builder.create()
+
+        evaluation_store_id = project.create_model_evaluation_store("output_model_evaluation").id
+    """
+
+    def __init__(self, name, project):
+        SingleOutputRecipeCreator.__init__(self, 'evaluation', name, project)
```
Comment (lpenet) [inline dataikuapi/dss/recipe.py:1353]: `SingleOutputRecipeCreator` => `DSSRecipeCreator`

File: dataikuapi/dss/recipe.py [hunk 3]
```diff
@@ -1313,6 +1314,60 @@ def with_input_model(self, model_id):
...
+        """Sets the ouput dataset containing the scored input"""
+        return self._with_output(name, role="main")
+
+    def with_output_metrics(self, name):
+        """Sets the output dataset containing the metrics"""
+        return self._with_output(name, role="metrics")
+
+    def with_output_evaluation_store(self, name):
+        """Sets the output dataset containing the metrics"""
+        return self._with_output(name, role="evalutationStore")
```
Comment (lpenet) [inline dataikuapi/dss/recipe.py:1369]: Typo: `evalutationStore` => `evaluationStore`.

**Review by lpenet — CHANGES_REQUESTED** [review]
Some little changes to make

File: dataikuapi/dss/recipe.py (hunk 3 above)
Comment (vthorey) [reply dataikuapi/dss/recipe.py]: 🤦

File: dataikuapi/dss/recipe.py [hunk 4]
```diff
@@ -1313,6 +1314,62 @@ def with_input_model(self, model_id):
...
+    """
+
+    def __init__(self, name, project):
+        DSSRecipeCreator.__init__(self, 'evaluation', name, project)
+
+    def with_input_model(self, model_id):
+        """Sets the input model"""
+        return self._with_input(model_id, self.project.project_key, "model")
+
+    def with_ouput(self, name):
```
Comment (FChataigner) [inline dataikuapi/dss/recipe.py:1360]: typo "with_ouput" -> "with_output"

**Review by FChataigner — CHANGES_REQUESTED** [review]

File: dataikuapi/dss/recipe.py (hunk 4 above)
Comment (vthorey) [reply dataikuapi/dss/recipe.py]: 🤦 that's fixed

**Comment by lpenet** [comment]
LGTM! Will validate it when backend part is fully implemented and tested.

**Comment by lpenet** [comment]
@instanceofme does not really add new entries to the API, rather extend existing ones. But nonetheless requesting your insight.

File: dataikuapi/dss/recipe.py [hunk 5]
```diff
@@ -1313,6 +1314,62 @@ def with_input_model(self, model_id):
...
+    def with_output(self, name):
+        """Sets the ouput dataset containing the scored input"""
+        return self._with_output(name, role="main")
+
+    def with_output_metrics(self, name):
+        """Sets the output dataset containing the metrics"""
+        return self._with_output(name, role="metrics")
+
+    def with_output_evaluation_store(self, name):
+        """Sets the output dataset containing the metrics"""
```
Comment (instanceofme) [inline dataikuapi/dss/recipe.py:1369]:
```suggestion
        """Sets the output model evaluation store"""
```

File: dataikuapi/dss/recipe.py [hunk 6]
```diff
@@ -1343,44 +1400,3 @@ def __init__(self, name, project):
     def with_input_model(self, model_id):
         """Sets the input model"""
         return self._with_input(model_id, self.project.project_key, "model")
-
-
-class DownloadRecipeCreator(SingleOutputRecipeCreator):
```
Comment (instanceofme) [inline dataikuapi/dss/recipe.py:1348]: Duplication to be handled in separate PR, see CH70141

**Review by instanceofme — APPROVED** [review]

**Pull request merged**